hypertrace-alert-engine/src/integrationTest/java/org/hypertrace/alert/engine/HypertraceAlertEngineTest.java

```diff
@@ -165,43 +167,95 @@ public static void setup() throws Exception {
+        .and("ATTRIBUTE_SERVICE_PORT_CONFIG", attributeService.getMappedPort(9012).toString())
+        .and("QUERY_SERVICE_HOST_CONFIG", queryService.getHost())
+        .and("QUERY_SERVICE_PORT_CONFIG", queryService.getMappedPort(8090).toString())
+        .and("ALERT_RULES_PATH", "build/resources/integrationTest/alert-rules-static-rule.json")
+        .and(
+            "NOTIFICATION_CHANNELS_PATH",
+            "build/resources/integrationTest/notification-channels-1.json")
+        .and("SERVICE_ADMIN_PORT", "10005")
+        .and("JOB_SUFFIX", "job1")
+        .execute(
+            () -> {
+              ConfigFactory.invalidateCaches();
+              IntegrationTestServerUtil.startServices(new String[] {"hypertrace-alert-engine"});
+            });
     // int x = 1;
     int retryCount = 0;
-    while (!(mockWebServer.getRequestCount() > 0) && retryCount++ < 10) {
+    while (!(mockWebServer.getRequestCount() == 1) && retryCount++ < 10) {
       Thread.sleep(Duration.ofSeconds(20).toMillis());
     }
-    LOG.info("Test completed. No of retries {}", retryCount);
     Assertions.assertTrue(retryCount < 10);
+    String notificationBody1 =
+        mockWebServer.takeRequest().getBody().readString(Charset.defaultCharset());
+    LOG.info(
+        "Static Threshold Test, No of retries {}, staticThreshold notification {}",
+        retryCount,
+        notificationBody1);
+
+    Assertions.assertTrue(notificationBody1.contains("static threshold"));
+    mockWebServer.close();
+  }
+
+  @Test
+  public void testDynamicThresholdNotificationSent() throws Exception {
+    LOG.info("Starting testDynamicThresholdNotificationSent");
+    MockWebServer mockWebServer = new MockWebServer();
+    mockWebServer.start(NOTIFICATION_CHANNEL_PORT2);
+    MockResponse mockedResponse =
+        new MockResponse().setResponseCode(200).addHeader("Content-Type", "application/json");
+    mockWebServer.enqueue(mockedResponse);
+
+    withEnvironmentVariable("ATTRIBUTE_SERVICE_HOST_CONFIG", attributeService.getHost())
+        .and("ATTRIBUTE_SERVICE_PORT_CONFIG", attributeService.getMappedPort(9012).toString())
+        .and("QUERY_SERVICE_HOST_CONFIG", queryService.getHost())
+        .and("QUERY_SERVICE_PORT_CONFIG", queryService.getMappedPort(8090).toString())
+        .and("ALERT_RULES_PATH", "build/resources/integrationTest/alert-rules-dynamic-rule.json")
+        .and(
+            "NOTIFICATION_CHANNELS_PATH",
+            "build/resources/integrationTest/notification-channels-2.json")
+        .and("SERVICE_ADMIN_PORT", "10010")
+        .and("JOB_SUFFIX", "job2")
+        .execute(
+            () -> {
+              ConfigFactory.invalidateCaches();
+              IntegrationTestServerUtil.startServices(new String[] {"hypertrace-alert-engine"});
+            });
+
+    Assertions.assertEquals(0, mockWebServer.getRequestCount());
+    // int x = 1;
+    int retryCount = 0;
+    // no notification sent
```

Comment:
Is this enhance later or is this expected that we don't see baseline notification?

Reply:
Yeah, this we will be enhanced later. Need multiple traces for this